test/jdk/java/net/DatagramSocket/DatagramSocketMulticasting.java

```diff
@@ -298,23 +304,33 @@ static void testSendReceive(DatagramSocket s, InetAddress group) throws IOExcept
         System.out.println("testSendReceive");
 
         // outgoing multicast interface needs to be set
-        assertTrue(s.getOption(IP_MULTICAST_IF) != null);
+        assertNotNull(s.getOption(IP_MULTICAST_IF));
 
         SocketAddress target = new InetSocketAddress(group, s.getLocalPort());
-        byte[] message = "hello".getBytes("UTF-8");
+        String message = "testSendReceive";
+        byte[] messageBytes = message.getBytes("UTF-8");
 
         // send message to multicast group
-        DatagramPacket p = new DatagramPacket(message, message.length);
+        DatagramPacket p = new DatagramPacket(messageBytes, messageBytes.length);
         p.setSocketAddress(target);
         s.send(p);
 
-        // receive message
+        // receive message with retry in case of stray messages
         s.setSoTimeout(0);
-        p = new DatagramPacket(new byte[1024], 100);
-        s.receive(p);
+        for (int i = 1; i <= MAX_TRIES; i++) {
+            p = new DatagramPacket(new byte[1024], 100);
+            s.receive(p);
+            String messageReceived = new String(p.getData(), 0, p.getLength());
+
+            System.out.println(String.format("TestSendReceive iteration [%s], Received DatagramPacket [%s] from [%s]", i, messageReceived, s.getLocalSocketAddress()));
+            if (s.getLocalPort() == p.getPort()) {
+                assertEquals(message, messageReceived,
+                        String.format("expected message %s, instead received %s%n", message, messageReceived));
+                break;
+            }
```

Comment:
Can you please re-format this to keep the lines at a sane line, it looks very messy to be adding 150+ lines to this test.